src/lib/evil/evil_libgen.c

```diff
@@ -0,0 +1,27 @@
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif /* HAVE_CONFIG_H */
+
+#include <evil_private.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+
+EVIL_API char *
+evil_basename(char *path)
+{
+   // Expected from posix
+   if ((path == NULL) || (*path == '\0') ) return ".";
+
+   // Remove trailing '\\'
+   unsigned int len = strlen(path) -1;
+   if ((path[len] == '/') || (path[len] == '\\')) path[len] = '\0';
+
+   // Search for the last slash
+   while(--len) if ((path[len] == '/') || (path[len] == '\\')) break;
```

Comment:
I really prefer a more explicit structuring. There was even 3 commands on the same line.

I'm gonna give my try here:

```suggestion
   when (len > 0)
     {
        --len;
        if ((path[len] == '/') || (path[len] == '\\')) break;
     }
```

Reply:
Oh, I am quite used to this type of while structures and didn't thought it could be messy. See if it is more clear now. 

Despite my bare metal C background I prefer to read a decrementing while-if-break structure than a minimalist for loop, but maybe it would be more understandable with a for? I don't know what these compilers likes the most though.
```C
for(; (path[len] == '/') || (path[len] == '\\'); --len);
```
What do you think?

Reply:
I'll address this in a later patch.